src/core/algorithms/cind/condition_miners/itemset_node.h

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+#include <algorithm>
+#include <list>
+#include <memory>
+#include <unordered_set>
+#include <vector>
+
+#include <boost/container_hash/hash.hpp>
+
+#include "item.h"
+
+namespace algos::cind {
+using BasketInfo = std::tuple<size_t, std::list<size_t>, bool>;
+
+class ItemsetNode : public std::enable_shared_from_this<ItemsetNode> {
+public:
+    ItemsetNode(std::shared_ptr<ItemsetNode> parent, Item value, std::list<BasketInfo> baskets_info,
+                double validity, double completeness)
+        : value_(std::move(value)),
+          baskets_info_(std::move(baskets_info)),
+          validity_(validity),
+          completeness_(completeness),
+          parent_node_(std::move(parent)) {}
+
+    std::shared_ptr<ItemsetNode> CreateChild(Item value, std::list<BasketInfo> baskets_info,
+                                             size_t included_baskets_cnt, double min_completeness) {
+        double included_contained_baskets_cnt = 0;
+        for (auto const& [_, real_id, is_included] : baskets_info) {
+            included_contained_baskets_cnt += is_included;
+        }
+        double completeness = included_contained_baskets_cnt / included_baskets_cnt;
+        if (completeness >= min_completeness) {
+            double validity = -1;
+            if (baskets_info.size()) {
+                validity = included_contained_baskets_cnt / baskets_info.size();
+            }
+            std::shared_ptr<ItemsetNode> new_element =
+                    std::make_shared<ItemsetNode>(shared_from_this(), std::move(value),
+                                                  std::move(baskets_info), validity, completeness);
```

Comment:
It would be better to use `auto` here.
```suggestion
            auto new_element =
                    std::make_shared<ItemsetNode>(shared_from_this(), std::move(value),
                                                  std::move(baskets_info), validity, completeness);
```